eng/pipelines/templates/stages/archetype-java-release-batch.yml

```diff
@@ -197,55 +197,55 @@ stages:
                         - Name: Azure/azure-sdk-for-java
                           Commitish: $(Build.SourceVersion)
                           WorkingDirectory: $(Pipeline.Workspace)/azure-sdk-for-java
+                  - pwsh: |
```

Comment:
As we chatted it would be best to try to do this artifact filter in the build job instead of in this release step.

Reply:
https://github.com/Azure/azure-sdk-for-java/blob/main/eng/scripts/Stage-MavenPackageArtifacts.ps1 is the place where this filtering should probably happen. This is the script we used for grouping the artifacts but it looks like we never factored in the artifact list itself and we assumed there was not anything else in the output.

Reply:
I moved the artifact shaping to Stage-MavenPackageArtifacts and its callers.    It was only invoked in 3 places, all yaml.  Each call passes -Artifacts as a filtered array using DevOps yml's convertToJson() and powershell's ConvertFrom-Json

Reply:
It was tested here: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1950192&view=results